### PR TITLE
Added ability to view and update the CKAN dataset resource URL via the edit form

### PR DESCRIPTION
**Pull request by jeremy-doghouse**
Hey @gollyg & @srowlands 

This guy solves ENV-590

Cheers!

### CODE REVIEW

File: modules/govcms_ckan_media/includes/govcms_ckan_media.field_config.inc [hunk 1]
```diff
@@ -126,15 +126,89 @@ function govcms_ckan_media_field_widget_form(&$form, &$form_state, $field, $inst
     $config = $file->{$field_name}[$langcode][$delta]['config'];
   }
 
+  // Hide preview as it fairly useless and always the same.
+  if (!empty($form['preview'])) {
+    $form['preview']['#access'] = FALSE;
+  }
```
Comment (gollyg) [inline modules/govcms_ckan_media/includes/govcms_ckan_media.field_config.inc:132]: Not sure if we should force hiding a default form element? Maybe we should, but wouldn't mind @srowlands second opinion on this.

File: modules/govcms_ckan_media/includes/govcms_ckan_media.field_config.inc [hunk 2]
```diff
@@ -170,16 +245,50 @@ function govcms_ckan_media_field_widget_form(&$form, &$form_state, $field, $inst
 }
 
 /**
- * Ajax callback for changing visualisations.
+ * Change the dataset url submit handler.
+ */
+function _govcms_ckan_media_field_widget_form_change_dataset_url($form, &$form_state) {
+  // Only proceed if explicitly set.
+  if (!$form_state['values']['dataset_url_change']) {
```
Comment (gollyg) [inline modules/govcms_ckan_media/includes/govcms_ckan_media.field_config.inc]: Should we use empty() here to guard against missing values, or are we certain the value will exist?

Comment (stooit) [reply modules/govcms_ckan_media/includes/govcms_ckan_media.field_config.inc]: 👍

File: modules/govcms_ckan_media/includes/govcms_ckan_media.field_config.inc (hunk 1 above)
Comment (stooit) [reply modules/govcms_ckan_media/includes/govcms_ckan_media.field_config.inc:132]: Yeah, I'm ok with this

Comment (gollyg) [reply modules/govcms_ckan_media/includes/govcms_ckan_media.field_config.inc:132]: Cool, let's leave as is.

File: modules/govcms_ckan_media/includes/govcms_ckan_media.field_config.inc (hunk 2 above)
Comment (gollyg) [reply modules/govcms_ckan_media/includes/govcms_ckan_media.field_config.inc]: @jeremy-doghouse can we add a guard condition here?

Comment (jeremy-doghouse) [reply modules/govcms_ckan_media/includes/govcms_ckan_media.field_config.inc]: @gollyg This submit handler gets added alongside the element so it should always exist (Line 163 & 203). That said I might cast to bool to make it more obvious.

**Pull request merged**